### PR TITLE
InfluxDB configuration files

### LINKED ISSUE #1038: Add ENV vars to config page
_Describe the issue here._

##### Relevant URLs
https://v2.docs.influxdata.com/v2.0/reference/cli/influxd/

> you can use env vars with INFLUXD as a prefix to any CLI flags and it will use that value, so `INFLUXD_QUERY_CONCURRENCY` could be used instead of `--query-concurrency` and so on

### LINKED ISSUE #1113: Add support for 'influxd' configuration files
PR: https://github.com/influxdata/influxdb/pull/18568

`influxd` now supports configuration files. Some important things to cover:

- Configuration setting order of precendence:
    1. CLI flags
    2. Environment variables
    3. Settings defined in the config file
- Syntax and structure of the config file (yaml, toml, json)
- All the configuration keys (same as the CLI flags without the `--` prefix)
- A config file can be provided via the `INFLUXD_CONFIG_FILE` env var. If a file is not provided via an env var, influxd will look in the current directory for a `config.yaml` file. If one does not exist, then it will continue unchanged.

##### Relevant URLs
- https://v2.docs.influxdata.com/v2.0/reference/config-options/
- https://v2.docs.influxdata.com/v2.0/reference/cli/influxd/

### LINKED ISSUE #1120: Update run docs for influxd v2
we now have the abililty to provide a config file in addition to env vars/ flags.

the usage changes in the PR may be helpful

##### Relevant URLs
- https://github.com/influxdata/influxdb/pull/18606

### PR DESCRIPTION
**Pull request by sanderson**
Closes #1038 
Closes #1113 
Closes #1120 

Restructured the Configuration options doc and updated the `influxd` docs to show alternate configuration methods:

1. CLI flags
2. Environment variables
3. A configuration file

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/v2.0/reference/config-options.md [hunk 1]
```diff
@@ -1,193 +1,696 @@
...
+When `influxd` starts, it checks for a **`config.yaml` in the current working directory**.
+To customize the file name or path of the configuration file, set the `INFLUXD_CONFIG_FILE`
+environment variable to the custom name or path.
+InfluxDB supports YAML, TOML, and JSON configuration files.
+
+##### Example configuration file
+{{< code-tabs-wrapper >}}
+{{% code-tabs %}}
+[YAML](#)
+[TOML](#)
```
Comment (jsteenb2) [inline content/v2.0/reference/config-options.md:37]: this is awesome 👏

File: layouts/shortcodes/cli/influxd-flags.md [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
+|      | `--session-length`             | TTL in minutes for newly created sessions (default `60`)                                                          | integer    | `INFLUXD_SESSION_LENGTH`                |
+|      | `--session-renew-disabled`     | Disables automatically extending session TTL on request                                                           |            | `INFLUXD_SESSION_RENEW_DISABLED`        |
+|      | `--store`                      | Data store for REST resources (bolt or memory) (default `bolt`)                                                   | string     | `INFLUXD_STORE`                         |
+|      | `--tls-cert`                   | Path to TLS certificate file                                                                                      | string     | `INFLUXD_TLS_CERT`                      |
+|      | `--tls-key`                    | Path to TLS private key file                                                                                      | string     | `INFLUXD_TLS_KEY`                       |
```
Comment (jsteenb2) [inline layouts/shortcodes/cli/influxd-flags.md:24]: very nice 👏, I see questions about this in the community every so often 👍

File: content/v2.0/reference/config-options.md [hunk 3]
```diff
@@ -1,193 +1,696 @@
 ---
 title: InfluxDB configuration options
 description: >
-  Configure InfluxDB when starting the `influxd` service.
-  This article outlines the different configuration options available.
+  Customize your InfluxDB configuration by passing configuration flags with the
```
Comment (russorat) [inline content/v2.0/reference/config-options.md:4]:
```suggestion
  Customize your InfluxDB configuration by passing configuration flags to the
```

File: content/v2.0/reference/config-options.md [hunk 4]
```diff
@@ -1,193 +1,696 @@
...
+related:
+  - /v2.0/reference/cli/influxd
 ---
 
+Customize your InfluxDB configuration by using configuration flags with the
+[`influxd` daemon](/v2.0/reference/cli/influxd/), setting environment variables,
+or defining configuration options in a configuration file.
+
+### Configuration precedence
+InfluxDB honors configurations settings using the following precedence:
```
Comment (russorat) [inline content/v2.0/reference/config-options.md:21]:
```suggestion
InfluxDB honors configuration settings using the following precedence:
```

File: content/v2.0/reference/config-options.md [hunk 5]
```diff
@@ -1,193 +1,696 @@
...
+
+1. `influxd` daemon flags
```
Comment (russorat) [inline content/v2.0/reference/config-options.md:23]: i mean, _technically_ `influxd` isn't a daemon unless you run it in the background. i feel like these are more commonly called command line flags or something

File: content/v2.0/reference/config-options.md [hunk 6]
```diff
@@ -1,193 +1,696 @@
...
+2. Environment variables
+3. Settings defined in a configuration file
+
+### InfluxDB configuration file
+When `influxd` starts, it checks for a **`config.yaml` in the current working directory**.
+To customize the file name or path of the configuration file, set the `INFLUXD_CONFIG_FILE`
```
Comment (russorat) [inline content/v2.0/reference/config-options.md:29]: Is there no way to set this via a command line flag? `influxd --config-path /path/to/my/config.yml`?

Comment (russorat) [reply content/v2.0/reference/config-options.md:29]: the 1.x version of this is INFLUXDB_CONFIG_PATH

Comment (sanderson) [reply content/v2.0/reference/config-options.md]: In talking with @jsteenb2, the presents the chicken and the egg problem.

Comment (jsteenb2) [reply content/v2.0/reference/config-options.md]: yah its a doozie. @russorat made a card for this. It'll likely require a larger refactor of our `cli` pkg to make this a reality. bit of a bummer. Was hoping to include this in the initial work but was just to much headache

File: content/v2.0/reference/config-options.md [hunk 7]
```diff
@@ -1,193 +1,698 @@
...
+Customize your InfluxDB configuration by using [`influxd`](/v2.0/reference/cli/influxd/)
+configuration flags, setting environment variables, or defining configuration
+options in a configuration file.
+
+### Configuration precedence
+InfluxDB honors configuration settings using the following precedence:
+
+1. `influxd` flags
+2. Environment variables
+3. Settings defined in a configuration file
```
Comment (kelseiv) [inline content/v2.0/reference/config-options.md:25]: Configuration file settings

File: content/v2.0/reference/config-options.md [hunk 8]
```diff
@@ -1,193 +1,698 @@
...
+
+### InfluxDB configuration file
+When `influxd` starts, it checks for a file named `config.*` (the file extension
+depends on the syntax) **in the current working directory**.
```
Comment (kelseiv) [inline content/v2.0/reference/config-options.md:29]: for example yaml, toml or json?

File: content/v2.0/reference/config-options.md [hunk 9]
```diff
@@ -1,193 +1,698 @@
...
+To customize the directory path of the configuration file, set the `INFLUXD_CONFIG_PATH`
```
Comment (kelseiv) [inline content/v2.0/reference/config-options.md:30]: link to instructions for how to do this?

File: layouts/shortcodes/cli/influxd-flags.md [hunk 10]
```diff
@@ -1,35 +1,35 @@
...
+|      | `--bolt-path`                  | Path to boltdb database (default `~/.influxdbv2/influxd.bolt`)                                                    | string     | `INFLUXD_BOLT_PATH`                     |
+|      | `--e2e-testing`                | Add /debug/flush endpoint to clear stores; used for end-to-end tests (default `false`)                            |            | `INFLUXD_E2E_TESTING`                   |
+|      | `--engine-path`                | Path to persistent engine files (default `~/.influxdbv2/engine`)                                                  | string     | `INFLUXD_ENGINE_PATH`                   |
+| `-h` | `--help`                       | Help for the `influxd` command                                                                                    |            |                                         |
+|      | `--http-bind-address`          | Bind address for the REST HTTP API (default `:9999`)                                                              | string     | `INFLUXD_HTTP_BIND_ADDRESSBIND_ADDRESS` |
```
Comment (kelseiv) [inline layouts/shortcodes/cli/influxd-flags.md:8]: extra "BIND_ADDRESS"?

Comment (sanderson) [reply layouts/shortcodes/cli/influxd-flags.md:8]: Oh, nice catch! Thanks!

**Pull request merged**